lib/mirage_block_safe.ml

```diff
@@ -41,6 +41,23 @@ let rec check_buffers op sector_size = function
     >>= fun () ->
     check_buffers op sector_size bs
 
+let check_in_range op size_sectors offset =
+  if offset < 0L || offset >= size_sectors
+  then fatalf "%s: sector offset out of range 0 <= %Ld < %Ld" op offset size_sectors
+  else Lwt.return (`Ok ())
+
+let check op sector_size size_sectors offset buffers =
+  let open Mirage_block_monad.Infix in
+  check_buffers op sector_size buffers
+  >>= fun () ->
+  check_in_range op size_sectors offset
+  >>= fun () ->
+  let length = List.fold_left (fun acc b -> Cstruct.len b + acc) 0 buffers in
+  let next_sector = Int64.(add offset (of_int @@ length / sector_size)) in
```

Comment:
If `offset` comes from the user, adding without checking for overflow might be a mistake. Might be better to subtract the buffer length from `size_sectors`.